src/modules/navigator/rtl_direct_mission_land.cpp

```diff
@@ -202,9 +232,171 @@ void RtlDirectMissionLand::setActiveMissionItems()
+										    next_position_mission_item.lat, next_position_mission_item.lon, &direction(0), &direction(1));
+
+							float hor_dist = get_distance_to_next_waypoint(hor_position_at_calculation_point(0),
+									 hor_position_at_calculation_point(1), next_position_mission_item.lat, next_position_mission_item.lon);
+
+							_rtl_time_estimator.addDistance(hor_dist, direction, 0.f);
+
+							// Add the vertical loiter
+							_rtl_time_estimator.addVertDistance(get_absolute_altitude_for_item(next_position_mission_item) -
+											    altitude_at_calculation_point);
+
+							break;
+						}
+
+					case NAV_CMD_LAND: // Fallthrough
+					case NAV_CMD_VTOL_LAND: {
+
+							matrix::Vector2f direction{};
+							get_vector_to_next_waypoint(hor_position_at_calculation_point(0), hor_position_at_calculation_point(1),
+										    next_position_mission_item.lat, next_position_mission_item.lon, &direction(0), &direction(1));
+
+							float hor_dist = get_distance_to_next_waypoint(hor_position_at_calculation_point(0),
+									 hor_position_at_calculation_point(1), next_position_mission_item.lat, next_position_mission_item.lon);
+
+							// For fixed wing, add diagonal line
+							if ((_vehicle_status_sub.get().vehicle_type != vehicle_status_s::VEHICLE_TYPE_FIXED_WING)
+							    && (!_vehicle_status_sub.get().is_vtol)) {
+
+
+								_rtl_time_estimator.addDistance(hor_dist, direction,
+												get_absolute_altitude_for_item(next_position_mission_item) - altitude_at_calculation_point);
+
+							} else {
+								// For VTOL, Rotary, go there horizontally first, then land
+								_rtl_time_estimator.addDistance(hor_dist, direction, 0.f);
+
+								_rtl_time_estimator.addDescendMCLand(get_absolute_altitude_for_item(next_position_mission_item) -
+												     altitude_at_calculation_point);
+							}
+
+							break;
+						}
+
+					default: {
+							// Default assume can go to the location directly
+							matrix::Vector2f direction{};
+							get_vector_to_next_waypoint(hor_position_at_calculation_point(0), hor_position_at_calculation_point(1),
+										    next_position_mission_item.lat, next_position_mission_item.lon, &direction(0), &direction(1));
+
+							float hor_dist = get_distance_to_next_waypoint(hor_position_at_calculation_point(0),
+									 hor_position_at_calculation_point(1), next_position_mission_item.lat, next_position_mission_item.lon);
+
+							_rtl_time_estimator.addDistance(hor_dist, direction,
+											get_absolute_altitude_for_item(next_position_mission_item) - altitude_at_calculation_point);
+							break;
+						}
+					}
+
+					start_item_index = next_mission_item_index + 1;
+					hor_position_at_calculation_point(0) = next_position_mission_item.lat;
+					hor_position_at_calculation_point(1) = next_position_mission_item.lon;
+					altitude_at_calculation_point = get_absolute_altitude_for_item(next_position_mission_item);
```

Comment:
That's not actually used or am I missing something?

Reply:
Which ones? The altitude and hor positions are used to calculate the distances to the next_position_mission_item. and the start_item_index to search for the next position item.

Reply:
I missed the while loop, never mind.